feat: support HashiCorp Vault with token based access (#38)
cluetec/lifeboat: samples/HashiCorp-Vault/README.md

# Sample: Backup HashiCorp Vault

> In this example we will show you how to back up an HashiCorp Vault instance.

Reviewer: ```suggestion
In this example we will show you how to backup an HashiCorp Vault instance.
```

## Requirements

The Vault instance needs to use [raft](https://developer.hashicorp.com/vault/docs/configuration/storage/raft) as the
underlying storage engine.

## Policy

In this sample we are using for simplicity reasons the "root token" to authorize us. In a real world scenario you

> would use a separate took or any other

Reviewer: ```suggestion
would use any other
```

[authentication method supported by vault](https://developer.hashicorp.com/vault/docs/auth).

**What's important here**: Normally your identity shouldn't have root permissions! The only permission you need for
creating the backup/snapshot is the following
([vault policy](https://developer.hashicorp.com/vault/docs/concepts/policies)):

```hcl
path "/sys/storage/raft/snapshot" {
  capabilities = ["read"]
}
```

## Run

### 1. Start the vault instance

The HashiCorp Vault setup is based on a docker compose setup.

```shell
# -d is starting the container in the background
$ docker-compose up -d
```

As we can't simply use the dev mode of Vault, we need to initialize and unseal it first. For this purpose the

> `docker-compose.yaml` contains next to the vault container an additional one which is called `vault-init`. This

Reviewer: ```suggestion
`docker-compose.yaml` contains an additional container which is called `vault-init`. This
```
